### PR TITLE
Adobe Stock Integration Overview and Usage documents

### LINKED ISSUE #82: Adobe Stock Integration overview
# New topic request

## Description

New topic for Adobe Stock Integration overview (project here: https://github.com/magento/adobe-stock-integration). 

## Content checklist

- [x] The topic provides an explanation of how the Adobe Stock Integration works.
- Create new .md file in src/catalog directory: adobe-stock.md
    - Topic title: "Adobe Stock Integration"
    - Topic contents:
        - [x] Beginning section to describe the integration features and requirements
        - [x] Provide information about licensing through Adobe Stock
        - Details for the two integration "Steps":
            - [x] "Create an IMS configuration"
            - [x] "Create Adobe Stock configuration in Magento"
        - Add link to html output ("Images and Video" section) in .yml files:
            - [x] `src/_data/ce/catalog.yml` -- add link after as the first child item below "Images and Videos"
            - [x] `src/_data/ee/catalog.yml` -- add link after as the first child item below "Images and Videos"
            - [x] `src/_data/b2b/catalog.yml` -- add link after as the first child item below "Images and Videos"

## Additional information/resources

Over related Adobe-specific articles on projects integrating Adobe Stock support that could be used as inspiration:

- https://docs.adobe.com/content/help/en/experience-manager-64/assets/using/aem-assets-adobe-stock.html
- https://helpx.adobe.com/experience-manager/6-5/assets/using/aem-assets-adobe-stock.html

This should be assigned to the [Adobe Stock Integration Docs Project](https://github.com/magento/merchdocs/projects/6) and stems from #77. I am happy to take this one.

### LINKED ISSUE #85: New topic for "Using Adobe Stock Images"
# New topic request

## Description

The topic provides an explanation of how image management in the Media Gallery works when combined with the Adobe Stock Integration extension. It should include an overview of:

- the Adobe Stock UI
- how signing in to Adobe Stock works
- how asset finding works between the Media Gallery as well as the Adobe Stock Integration
- how images can be saved from Adobe Stock into Magento's Media Gallery
- how licensing images work with Adobe Stock

## Content checklist

- [ ] Create new .md file in src/catalog directory: adobe-stock-manage.md
- [ ] Topic title: "Using Adobe Stock Images"
- Topic contents:
    - [ ] Overview of the Adobe Stock UI as displayed in Magento (do not use numbered or lettered callouts as in the AEM docs)
    - [ ] "Account sign in" section
    - [ ] "Find assets" section
    - [ ] "Add to Media Gallery" section
    - [ ] "License assets"  section
- Add link to html output ("Images and Video" section) in .yml files:
    - [ ] `src/_data/ce/catalog.yml` -- add link as child of "Adding Product Images"
    - [ ] `src/_data/ee/catalog.yml` -- add link as child of "Adding Product Images"
    - [ ] `src/_data/b2b/catalog.yml` -- add link as child of "Adding Product Images"

## Additional information/resources

This should be assigned to the [Adobe Stock Integration Docs Project](https://github.com/magento/merchdocs/projects/6) and stems from #77. I am happy to take this one.

### PR DESCRIPTION
**Pull request by filmaj**
## Purpose of this pull request

This pull request adds high-level information about the Adobe Stock Integration, what it accomplishes, what features it provides and how to use them, what requirements it has and how to configure the integration.

## Affected documentation pages

- The main menu on the left: this overview document is added as a child menu item under the Media Gallery
- The new overview document itself
- New usage documentation

## Affected Magento editions

- [x] Open Source
- [x] Commerce
- [x] B2B

## Links to Magento source code or PRs

- The integration itself is revisioned in https://github.com/magento/adobe-stock-integration

## Additional information

This fixes #82, this fixes #85.

FYI @sivaschenko @diazwatson @konarshankar07

Full overview article screenshot for those not wanting to build the docs site 😄: 

127 0 0 1_4000_m2_ce_user_guide_cms_adobe-stock html

Full usage article screenshot:

localhost_4000_m2_ce_user_guide_cms_adobe-stock-manage html

### CODE REVIEW

**Comment by devops-devdocs** [comment]
An admin must run tests on this PR before it can be merged.

**Comment by filmaj** [comment]
Hmm, I think I put together too much information in this article. The usage instructions should be split out into a separate topic and addressed via #85 - I will work on doing that.

**Comment by filmaj** [comment]
Now that I'm looking at this with a wider view, I also don't think this belongs as a menu item under the Catalog / Products section. If anything, it should belong under the CMS/Content/Media Storage, since the Integration is functionality serving that area of Magento. Will update this PR accordingly.

**Comment by filmaj** [comment]
OK, I think this is ready for review next. I'll put together the usage instructions next.

**Comment by filmaj** [comment]
I decided to bundle both configuration and usage articles in this one PR, as they are pretty dependent on each other.

**Comment by dshevtsov** [comment]
@filmaj the images seem to be pretty heavy. Please optimize them by running `rake check path=src/images/images/adobe-stock`.

For future contributions, consider running `rake check` before you commit any images. We need this, because binary files inflate the repository rapidly.

**Comment by filmaj** [comment]
Thanks @dshevtsov ! I have ran that command, squashed and force-pushed back up to my fork, so hopefully that won't negatively affect the repo size.

File: src/_data/b2b/toc/content.yml [hunk 1]
```diff
@@ -34,6 +34,12 @@ pages:
         url: "/cms/page-home-new.html"
     - label: Media Storage
       url: "/cms/media-storage.html"
+      chidlren:
```
Comment (konarshankar07) [inline src/_data/b2b/toc/content.yml]: Please fix typo of `children`

File: src/_data/ee/toc/content.yml [hunk 2]
```diff
@@ -34,6 +34,12 @@ pages:
         url: "/cms/page-home-new.html"
     - label: Media Storage
       url: "/cms/media-storage.html"
+      chidlren:
```
Comment (konarshankar07) [inline src/_data/ee/toc/content.yml]: Please fix typo of children

File: src/cms/adobe-stock-manage.md [hunk 3]
```diff
@@ -0,0 +1,192 @@
...
+![Adobe Stock Search Results for the "mango" keyword]({% link images/images/adobe-stock/mango.png %}){: .zoom}
+
+#### Filter Adobe Stock Assets
+
+1. [Run a keyword search for Adobe Stock assets][search-by-keywords]
+2. Click the **Filters** button
+3. You have several filters available to you to further refine your search
+   results:
+
+    |Filter|Description|
```
Comment (konarshankar07) [inline src/cms/adobe-stock-manage.md]: I think we can also add another column for the default value which is selected in the filter i.e safe search is enabled by default

File: src/cms/adobe-stock.md [hunk 4]
```diff
@@ -0,0 +1,81 @@
...
+        - `${HOST}` is your Magento instance's fully qualified domain name (i.e.
+          `https://store.myshop.com`)
+        - `${ADMIN_URI}` is your instance's admin URI which can be retrieved by running
+          `magento info:adminuri` (i.e.: `admin_hgkq1l`)
+    - **Redirect URI pattern**: The redirect URI pattern is the same as your redirect
+      URI with two changes:
+        - Any periods (`.`) need to be escaped with two backslashes (`\\`)
+        - You will need to add `.*` to the end of the pattern
+        - Using the example from the Default redirect URI field above, the
+          Redirect URI pattern would be `https://store\\.myshop\\.com/admin_hgkq1l/adobe_ims/oauth/callback/.*`
```
Comment (konarshankar07) [inline src/cms/adobe-stock.md]: I think this is complicated step from merchant point of view and we can simplify by merchant to copy and paste the `Default redirect URI` and `Redirect URI pattern` from Magento system configuration

Comment (leslietilling) [reply src/cms/adobe-stock.md]: You can use the formatting for a fenced code block that can be easily copied if needed.

File: src/_data/b2b/toc/content.yml (hunk 1 above)
Comment (leslietilling) [reply src/_data/b2b/toc/content.yml]: I will fix after merge

File: src/cms/adobe-stock.md (hunk 4 above)
Comment (leslietilling) [reply src/cms/adobe-stock.md]: I will work on this formatting after merge

**Pull request merged**